### PR TITLE
Koshan/user favourite endpoints

### PR DESCRIPTION
**Pull request by Koshan-Samarasinghe**
Dashboard widgets Apis for favorite billers, payees and maintenance.

### CODE REVIEW

File: web/src/main/java/lk/directpay/sampathvishwa/web/config/SecurityConfig.java [hunk 1]
```diff
@@ -57,6 +57,7 @@ public SecurityFilterChain filterChain(HttpSecurity http, JwtRequestFilter jwtRe
                 request.requestMatchers("/api/user/**").permitAll()
                         .requestMatchers("/api/resource/status").permitAll()
                         .requestMatchers("/forgot-password/**").permitAll()
+                        .requestMatchers("/api/dashboard/**").permitAll()
```
Comment (pramitha1166) [inline web/src/main/java/lk/directpay/sampathvishwa/web/config/SecurityConfig.java:60]: dashboard api should be autheticated. remove this line.

File: web/src/main/java/lk/directpay/sampathvishwa/web/controller/DashboardController.java [hunk 2]
```diff
@@ -2,18 +2,52 @@
...
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/api/dashboard")
 public class DashboardController {
 
+    private final DashboardServiceImpl dashboardServiceImpl;
```
Comment (pramitha1166) [inline web/src/main/java/lk/directpay/sampathvishwa/web/controller/DashboardController.java:13]: Dont call impplementation class. call interface

File: web/src/main/java/lk/directpay/sampathvishwa/web/config/SecurityConfig.java (hunk 1 above)
Comment (Koshan-Samarasinghe) [reply web/src/main/java/lk/directpay/sampathvishwa/web/config/SecurityConfig.java]: Removed.

File: web/src/main/java/lk/directpay/sampathvishwa/web/controller/DashboardController.java (hunk 2 above)
Comment (Koshan-Samarasinghe) [reply web/src/main/java/lk/directpay/sampathvishwa/web/controller/DashboardController.java]: Fixed.

**Pull request merged**